lib/takeaway.rb

```diff
@@ -0,0 +1,81 @@
+require './lib/menu.rb'
+require 'twilio-ruby'
+require './auth_credentials'
+
+class TakeAway
+  attr_reader :basket, :menu, :total
+
+  def initialize(menu: Menu.new)
+    @basket = []
+    @menu = menu
+    @total = 0
+  end
+
+  def read_menu
+    @menu.read_menu
+  end
+
+  def add_to_basket(dessert)
+    raise 'Please pick a dessert from the menu.' unless in_menu?(dessert)
+    adding_to_total(dessert)
+  end
+
+  def remove_from_basket(dessert)
```

Comment:
It would be easier to calculate the total only when you need to return it. This way it would help you delete this code here